src/randoop/util/OneMoreElementList.java

```diff
@@ -45,4 +45,42 @@ public String toString() {
     return toJDKList().toString();
   }
 
+  @Override
+  public Iterator<T> iterator() {
+    return new OMEIterator(list.iterator(), lastElement);
+  }
+
+  private class OMEIterator implements Iterator<T> {
```

Comment:
Stands for "one more element"?  Please document.  I'm inclined to implement this by calling something else (such as plume-lib's MergedIterator), even though it's less efficient, because that other code is probably better-tested.